backend/models/user.js

```diff
@@ -0,0 +1,45 @@
+var mongoose = require('mongoose');
+var bcrypt = require('bcrypt');
+var UserSchema = new mongoose.Schema({
+    email: {type: String, unique: true, required: true, trim: true},
+    name: {type: String, required: true, trim: true},
+    password: {type: String, required: true},
+    posts: [{type: mongoose.Schema.Types.ObjectId, ref: 'Post'}],
+    comments: [{type: mongoose.Schema.Types.ObjectId, ref: 'Comment'}],
+    imagePath: String
+});
+// authenticate input against database documents
+UserSchema.statics.authenticate = function(email, password, callback) {
+    User.findOne({email: email})
+        .exec(function (error, user) {
+            if (error) {
+                return callback(error);
+            }
+            else if (!user) {
+                var err = new Error('User not found.');
+                err.status = 401;
```

Comment:
You probably want a more semantically correct error code like 400 bad requests or 404 not found